
focused.csv.gz and focused.T.csv.gz are rounded to 1000.


In [1]:
import numpy as np
print(np.__version__)

1.19.5


In [2]:
import pandas as pd
print(pd.__version__)

1.1.5


In [3]:
import matplotlib
matplotlib.__version__

'3.3.4'

In [4]:
import matplotlib.pyplot as plt

In [5]:
# Adjust plot size
#options(repr.plot.width=16, repr.plot.height=6)

# For some reason, if this is in the same cell as the import command, it doesn't work.
plt.rcParams["figure.figsize"] = [18.0,8.0]

In [6]:
max_rows=pd.get_option('display.max_rows')

# Read Metadata 1

In [7]:
mt1=pd.read_csv("TCGA/metadata.cart.TCGA.GBM-LGG.WGS.bam.2020-07-17.csv",header=None,
 names=['submitter','days_to_death','gender','disease_type','center_name'])

In [8]:
mt1.shape

(127, 5)

In [9]:
mt1.rename(columns={"submitter": "subject"},inplace=True)
mt1.rename(columns={"gender": "gender1"},inplace=True)

In [10]:
mt1.head()

subject  days_to_death gender1             disease_type  \
0  02-2483            NaN    male  Glioblastoma Multiforme   
1  02-2485            NaN    male  Glioblastoma Multiforme   
2  06-0125         1448.0  female  Glioblastoma Multiforme   
3  06-0152          375.0    male  Glioblastoma Multiforme   
4  06-0157           97.0  female  Glioblastoma Multiforme   

                          center_name  
0  Broad Institute of MIT and Harvard  
1  Broad Institute of MIT and Harvard  
2  Broad Institute of MIT and Harvard  
3  Broad Institute of MIT and Harvard  
4  Broad Institute of MIT and Harvard

In [11]:
list(mt1)

['subject', 'days_to_death', 'gender1', 'disease_type', 'center_name']

# Read Metadata 2

In [12]:
mt2=pd.read_csv("TCGA/TCGA.Glioma.metadata.tsv",sep="\t")

In [13]:
mt2.shape

(1114, 18)

In [14]:
mt2.head()

case_submitter_id project_id primary_diagnosis   race  \
0      TCGA-02-0001   TCGA-GBM      Glioblastoma  white   
1      TCGA-02-0003   TCGA-GBM      Glioblastoma  white   
2      TCGA-02-0004   TCGA-GBM      Glioblastoma  white   
3      TCGA-02-0006   TCGA-GBM      Glioblastoma  white   
4      TCGA-02-0007   TCGA-GBM      Glioblastoma  white   

                ethnicity  gender RE_names IDH     x1p19q TERT  \
0  not hispanic or latino  female      NaN  WT  non-codel  NaN   
1  not hispanic or latino    male      NaN  WT  non-codel  NaN   
2  not hispanic or latino    male      NaN  WT        NaN  NaN   
3  not hispanic or latino  female      NaN  WT  non-codel  NaN   
4  not hispanic or latino  female      NaN  WT  non-codel  NaN   

         IDH_1p19q_status                           WHO_groups  \
0  IDH-WT:1p19q-non-codel  glioblastoma:IDH-WT:1p19q-non-codel   
1  IDH-WT:1p19q-non-codel  glioblastoma:IDH-WT:1p19q-non-codel   
2         IDH-WT:1p19q-NA                                  NaN   
3  IDH-WT:1p19q-non-codel  glioblastoma:IDH-WT:1p19q-non-codel   
4  IDH-WT:1p19q-non-codel  glioblastoma:IDH-WT:1p19q-non-codel   

                     Triple_group     Tissue_sample_location          MGMT  \
0  IDH-WT:1p19q-non-codel:TERT-NA  MD Anderson Cancer Center  Unmethylated   
1  IDH-WT:1p19q-non-codel:TERT-NA  MD Anderson Cancer Center  Unmethylated   
2         IDH-WT:1p19q-NA:TERT-NA  MD Anderson Cancer Center           NaN   
3  IDH-WT:1p19q-non-codel:TERT-NA  MD Anderson Cancer Center  Unmethylated   
4  IDH-WT:1p19q-non-codel:TERT-NA  MD Anderson Cancer Center  Unmethylated   

    Age  Survival_months  Vital_status  
0  44.0        11.762054           1.0  
1  50.0         4.731106           1.0  
2  59.0        11.334941           1.0  
3  56.0        18.333034           1.0  
4  40.0        23.162705           1.0

In [15]:
list(mt2)

['case_submitter_id',
 'project_id',
 'primary_diagnosis',
 'race',
 'ethnicity',
 'gender',
 'RE_names',
 'IDH',
 'x1p19q',
 'TERT',
 'IDH_1p19q_status',
 'WHO_groups',
 'Triple_group',
 'Tissue_sample_location',
 'MGMT',
 'Age',
 'Survival_months',
 'Vital_status']

In [16]:
mt2['subject']=mt2['case_submitter_id'].astype(str).str[5:12]

In [17]:
mt2[['subject','case_submitter_id']].head()

subject case_submitter_id
0  02-0001      TCGA-02-0001
1  02-0003      TCGA-02-0003
2  02-0004      TCGA-02-0004
3  02-0006      TCGA-02-0006
4  02-0007      TCGA-02-0007

In [18]:
mt2.drop('case_submitter_id',axis='columns',inplace=True)
mt2.rename(columns={"gender": "gender2"},inplace=True)

In [19]:
list(mt2)

['project_id',
 'primary_diagnosis',
 'race',
 'ethnicity',
 'gender2',
 'RE_names',
 'IDH',
 'x1p19q',
 'TERT',
 'IDH_1p19q_status',
 'WHO_groups',
 'Triple_group',
 'Tissue_sample_location',
 'MGMT',
 'Age',
 'Survival_months',
 'Vital_status',
 'subject']

# Merge Metadata

In [20]:
#meta=mt2.merge(mt1,left_on='subject2',right_on='subject1',suffixes=["_mt2","_mt1"],how='inner')
meta=mt2.merge(mt1,on='subject',suffixes=["_mt2","_mt1"],how='inner')

In [21]:
del mt1
del mt2

In [22]:
meta.head()

project_id primary_diagnosis                       race  \
0   TCGA-GBM      Glioblastoma                      asian   
1   TCGA-GBM      Glioblastoma  black or african american   
2   TCGA-GBM      Glioblastoma                      white   
3   TCGA-GBM      Glioblastoma                      white   
4   TCGA-GBM      Glioblastoma                      white   

                ethnicity gender2                   RE_names     IDH  \
0  not hispanic or latino    male  02-2483-01A-01R-1849-01+1  Mutant   
1  not hispanic or latino    male  02-2485-01A-01R-1849-01+1      WT   
2  not hispanic or latino  female  06-0125-01A-01R-1849-01+1      WT   
3  not hispanic or latino    male                        NaN      WT   
4  not hispanic or latino  female  06-0157-01A-01R-1849-01+1      WT   

      x1p19q    TERT            IDH_1p19q_status  ...  \
0  non-codel      WT  IDH-Mutant:1p19q-non-codel  ...   
1  non-codel  Mutant      IDH-WT:1p19q-non-codel  ...   
2  non-codel  Mutant      IDH-WT:1p19q-non-codel  ...   
3  non-codel     NaN      IDH-WT:1p19q-non-codel  ...   
4  non-codel  Mutant      IDH-WT:1p19q-non-codel  ...   

      Tissue_sample_location          MGMT   Age Survival_months  \
0  MD Anderson Cancer Center    Methylated  43.0       15.310383   
1  MD Anderson Cancer Center  Unmethylated  53.0       15.441803   
2        Henry Ford Hospital    Methylated  63.0       47.573895   
3        Henry Ford Hospital           NaN  68.0       12.320588   
4        Henry Ford Hospital           NaN  63.0        3.186925   

   Vital_status  subject  days_to_death gender1             disease_type  \
0           0.0  02-2483            NaN    male  Glioblastoma Multiforme   
1           0.0  02-2485            NaN    male  Glioblastoma Multiforme   
2           1.0  06-0125         1448.0  female  Glioblastoma Multiforme   
3           1.0  06-0152          375.0    male  Glioblastoma Multiforme   
4           1.0  06-0157           97.0  female  Glioblastoma Multiforme   

                          center_name  
0  Broad Institute of MIT and Harvard  
1  Broad Institute of MIT and Harvard  
2  Broad Institute of MIT and Harvard  
3  Broad Institute of MIT and Harvard  
4  Broad Institute of MIT and Harvard  

[5 rows x 22 columns]

In [23]:
meta.drop(meta[(meta['subject']=='CS-6186') & (meta['center_name']=='Harvard Medical School')].index,inplace=True)

In [24]:
meta.drop(meta[(meta['subject']=='DU-5872') & (meta['center_name']=='Harvard Medical School')].index,inplace=True)

In [25]:
meta.columns

Index(['project_id', 'primary_diagnosis', 'race', 'ethnicity', 'gender2',
       'RE_names', 'IDH', 'x1p19q', 'TERT', 'IDH_1p19q_status', 'WHO_groups',
       'Triple_group', 'Tissue_sample_location', 'MGMT', 'Age',
       'Survival_months', 'Vital_status', 'subject', 'days_to_death',
       'gender1', 'disease_type', 'center_name'],
      dtype='object')

In [26]:
meta['TERT'].value_counts()

Mutant    68
WT        54
Name: TERT, dtype: int64

In [27]:
meta['primary_diagnosis'].value_counts()

Glioblastoma                     35
Oligodendroglioma, NOS           33
Astrocytoma, anaplastic          24
Mixed glioma                     16
Astrocytoma, NOS                  9
Oligodendroglioma, anaplastic     8
Name: primary_diagnosis, dtype: int64

In [28]:
meta['IDH'].value_counts()

Mutant    74
WT        50
Name: IDH, dtype: int64

In [29]:
meta['x1p19q'].value_counts()

non-codel    102
codel         23
Name: x1p19q, dtype: int64

## We need a multi-level index to merge with df later so adding an empty index level

In [30]:
meta.columns=[meta.columns,[''] * len(meta.columns)]

In [31]:
#meta.columns.names=['chromosome','position']

In [32]:
meta.columns

MultiIndex([(            'project_id', ''),
            (     'primary_diagnosis', ''),
            (                  'race', ''),
            (             'ethnicity', ''),
            (               'gender2', ''),
            (              'RE_names', ''),
            (                   'IDH', ''),
            (                'x1p19q', ''),
            (                  'TERT', ''),
            (      'IDH_1p19q_status', ''),
            (            'WHO_groups', ''),
            (          'Triple_group', ''),
            ('Tissue_sample_location', ''),
            (                  'MGMT', ''),
            (                   'Age', ''),
            (       'Survival_months', ''),
            (          'Vital_status', ''),
            (               'subject', ''),
            (         'days_to_death', ''),
            (               'gender1', ''),
            (          'disease_type', ''),
            (           'center_name', '')],
           )

# Read raw data and cleanup

In [33]:
#df=pd.read_csv("focused.csv.gz",header=[0,1,2,3],index_col=[0,1])
#df.columns.names=['sample','pup','hkle','q']
#df=df.T
df=pd.read_csv("focused.T.csv.gz",header=[0,1],index_col=[0,1,2,3])

df.reset_index(inplace=True)
df.head()


chromosome       sample     pup      hkle mapq    chr1                  \
position                                       1048000 1190000 1308000   
0           02-2483-01A  paired  HERVK113  Q20       0       0       0   
1           02-2483-01A  paired     SVA_A  Q20       0       0       0   
2           02-2483-01A  paired     SVA_B  Q20       0       0       0   
3           02-2483-01A  paired     SVA_C  Q20       0       0       0   
4           02-2483-01A  paired     SVA_D  Q20       0       0       0   

chromosome                          ...     chrY                             \
position   1409000 1410000 1580000  ... 19535000 19536000 19699000 19700000   
0                4       1       4  ...        0        0        2        0   
1                0       0       0  ...        0        0        0        0   
2                0       0       0  ...        0        0        0        0   
3                0       0       0  ...        0        0        0        0   
4                0       0       0  ...        0        0        0        0   

chromosome                                                        
position   20552000 20553000 20976000 21047000 21217000 21639000  
0                 0        0        0        0        0        0  
1                 0        0        0        0        0        0  
2                 0        0        0        0        0        0  
3                 0        0        0        0        0        0  
4                 0        0        0        0        0        0  

[5 rows x 17413 columns]

In [34]:
df['subject']=df['sample'].str[0:7]
df['subject']

0       02-2483
1       02-2483
2       02-2483
3       02-2483
4       02-2483
         ...   
1913    TQ-A8XE
1914    TQ-A8XE
1915    TQ-A8XE
1916    TQ-A8XE
1917    TQ-A8XE
Name: subject, Length: 1918, dtype: object

In [35]:
df.columns

MultiIndex([( 'sample',         ''),
            (    'pup',         ''),
            (   'hkle',         ''),
            (   'mapq',         ''),
            (   'chr1',  '1048000'),
            (   'chr1',  '1190000'),
            (   'chr1',  '1308000'),
            (   'chr1',  '1409000'),
            (   'chr1',  '1410000'),
            (   'chr1',  '1580000'),
            ...
            (   'chrY', '19536000'),
            (   'chrY', '19699000'),
            (   'chrY', '19700000'),
            (   'chrY', '20552000'),
            (   'chrY', '20553000'),
            (   'chrY', '20976000'),
            (   'chrY', '21047000'),
            (   'chrY', '21217000'),
            (   'chrY', '21639000'),
            ('subject',         '')],
           names=['chromosome', 'position'], length=17414)

In [36]:
df.head()

chromosome       sample     pup      hkle mapq    chr1                  \
position                                       1048000 1190000 1308000   
0           02-2483-01A  paired  HERVK113  Q20       0       0       0   
1           02-2483-01A  paired     SVA_A  Q20       0       0       0   
2           02-2483-01A  paired     SVA_B  Q20       0       0       0   
3           02-2483-01A  paired     SVA_C  Q20       0       0       0   
4           02-2483-01A  paired     SVA_D  Q20       0       0       0   

chromosome                          ...     chrY                             \
position   1409000 1410000 1580000  ... 19536000 19699000 19700000 20552000   
0                4       1       4  ...        0        2        0        0   
1                0       0       0  ...        0        0        0        0   
2                0       0       0  ...        0        0        0        0   
3                0       0       0  ...        0        0        0        0   
4                0       0       0  ...        0        0        0        0   

chromosome                                               subject  
position   20553000 20976000 21047000 21217000 21639000           
0                 0        0        0        0        0  02-2483  
1                 0        0        0        0        0  02-2483  
2                 0        0        0        0        0  02-2483  
3                 0        0        0        0        0  02-2483  
4                 0        0        0        0        0  02-2483  

[5 rows x 17414 columns]

In [37]:
df.set_index(['subject','sample','pup','hkle','mapq'],inplace=True)

In [38]:
df.head()

chromosome                                  chr1                          \
position                                 1048000 1190000 1308000 1409000   
subject sample      pup    hkle     mapq                                   
02-2483 02-2483-01A paired HERVK113 Q20        0       0       0       4   
                           SVA_A    Q20        0       0       0       0   
                           SVA_B    Q20        0       0       0       0   
                           SVA_C    Q20        0       0       0       0   
                           SVA_D    Q20        0       0       0       0   

chromosome                                                                \
position                                 1410000 1580000 1581000 2211000   
subject sample      pup    hkle     mapq                                   
02-2483 02-2483-01A paired HERVK113 Q20        1       4       0       0   
                           SVA_A    Q20        0       0       0       0   
                           SVA_B    Q20        0       0       0       0   
                           SVA_C    Q20        0       0       0       0   
                           SVA_D    Q20        0       0       0       0   

chromosome                                                ...     chrY  \
position                                 2513000 2581000  ... 19535000   
subject sample      pup    hkle     mapq                  ...            
02-2483 02-2483-01A paired HERVK113 Q20        0       0  ...        0   
                           SVA_A    Q20        0       0  ...        0   
                           SVA_B    Q20        0       0  ...        0   
                           SVA_C    Q20        0       0  ...        0   
                           SVA_D    Q20        0       0  ...        0   

chromosome                                                                    \
position                                 19536000 19699000 19700000 20552000   
subject sample      pup    hkle     mapq                                       
02-2483 02-2483-01A paired HERVK113 Q20         0        2        0        0   
                           SVA_A    Q20         0        0        0        0   
                           SVA_B    Q20         0        0        0        0   
                           SVA_C    Q20         0        0        0        0   
                           SVA_D    Q20         0        0        0        0   

chromosome                                                                    \
position                                 20553000 20976000 21047000 21217000   
subject sample      pup    hkle     mapq                                       
02-2483 02-2483-01A paired HERVK113 Q20         0        0        0        0   
                           SVA_A    Q20         0        0        0        0   
                           SVA_B    Q20         0        0        0        0   
                           SVA_C    Q20         0        0        0        0   
                           SVA_D    Q20         0        0        0        0   

chromosome                                         
position                                 21639000  
subject sample      pup    hkle     mapq           
02-2483 02-2483-01A paired HERVK113 Q20         0  
                           SVA_A    Q20         0  
                           SVA_B    Q20         0  
                           SVA_C    Q20         0  
                           SVA_D    Q20         0  

[5 rows x 17409 columns]

In [39]:
df.shape

(1918, 17409)

## Remove "pup" and "q" as they are useless in this dataset.

In [40]:
df=df.droplevel('pup',axis='index').droplevel('mapq',axis='index')

In [41]:
df.head()

chromosome                      chr1                                          \
position                     1048000 1190000 1308000 1409000 1410000 1580000   
subject sample      hkle                                                       
02-2483 02-2483-01A HERVK113       0       0       0       4       1       4   
                    SVA_A          0       0       0       0       0       0   
                    SVA_B          0       0       0       0       0       0   
                    SVA_C          0       0       0       0       0       0   
                    SVA_D          0       0       0       0       0       0   

chromosome                                                    ...     chrY  \
position                     1581000 2211000 2513000 2581000  ... 19535000   
subject sample      hkle                                      ...            
02-2483 02-2483-01A HERVK113       0       0       0       0  ...        0   
                    SVA_A          0       0       0       0  ...        0   
                    SVA_B          0       0       0       0  ...        0   
                    SVA_C          0       0       0       0  ...        0   
                    SVA_D          0       0       0       0  ...        0   

chromosome                                                                 \
position                     19536000 19699000 19700000 20552000 20553000   
subject sample      hkle                                                    
02-2483 02-2483-01A HERVK113        0        2        0        0        0   
                    SVA_A           0        0        0        0        0   
                    SVA_B           0        0        0        0        0   
                    SVA_C           0        0        0        0        0   
                    SVA_D           0        0        0        0        0   

chromosome                                                        
position                     20976000 21047000 21217000 21639000  
subject sample      hkle                                          
02-2483 02-2483-01A HERVK113        0        0        0        0  
                    SVA_A           0        0        0        0  
                    SVA_B           0        0        0        0  
                    SVA_C           0        0        0        0  
                    SVA_D           0        0        0        0  

[5 rows x 17409 columns]

In [42]:
df.shape

(1918, 17409)

## Drop recurrent tumor (-02?)

In [43]:
df.loc[df.index.get_level_values('sample').str.contains("-02.$")].index

MultiIndex([('06-0125', '06-0125-02A', 'HERVK113'),
            ('06-0125', '06-0125-02A',    'SVA_A'),
            ('06-0125', '06-0125-02A',    'SVA_B'),
            ('06-0125', '06-0125-02A',    'SVA_C'),
            ('06-0125', '06-0125-02A',    'SVA_D'),
            ('06-0125', '06-0125-02A',    'SVA_E'),
            ('06-0125', '06-0125-02A',    'SVA_F'),
            ('06-0152', '06-0152-02A', 'HERVK113'),
            ('06-0152', '06-0152-02A',    'SVA_A'),
            ('06-0152', '06-0152-02A',    'SVA_B'),
            ...
            ('TQ-A7RV', 'TQ-A7RV-02A',    'SVA_D'),
            ('TQ-A7RV', 'TQ-A7RV-02A',    'SVA_E'),
            ('TQ-A7RV', 'TQ-A7RV-02A',    'SVA_F'),
            ('TQ-A8XE', 'TQ-A8XE-02A', 'HERVK113'),
            ('TQ-A8XE', 'TQ-A8XE-02A',    'SVA_A'),
            ('TQ-A8XE', 'TQ-A8XE-02A',    'SVA_B'),
            ('TQ-A8XE', 'TQ-A8XE-02A',    'SVA_C'),
            ('TQ-A8XE', 'TQ-A8XE-02A',    'SVA_D'),
            ('TQ-A8XE', 'TQ-A8XE-02A',    'SVA_E

In [44]:
df.drop(df.loc[df.index.get_level_values('sample').str.contains("-02.$")].index,axis='index',inplace=True)

In [45]:
df.loc[df.index.get_level_values('sample').str.contains("-02.$")].index

MultiIndex([], names=['subject', 'sample', 'hkle'])

## Drop solid normal (-11?) (I've seen a couple)

In [46]:
df.loc[df.index.get_level_values('sample').str.contains("-11.$")].index

MultiIndex([('FG-5963', 'FG-5963-11A', 'HERVK113'),
            ('FG-5963', 'FG-5963-11A',    'SVA_A'),
            ('FG-5963', 'FG-5963-11A',    'SVA_B'),
            ('FG-5963', 'FG-5963-11A',    'SVA_C'),
            ('FG-5963', 'FG-5963-11A',    'SVA_D'),
            ('FG-5963', 'FG-5963-11A',    'SVA_E'),
            ('FG-5963', 'FG-5963-11A',    'SVA_F')],
           names=['subject', 'sample', 'hkle'])

In [47]:
df.drop(df.loc[df.index.get_level_values('sample').str.contains("-11.$")].index,axis='index',inplace=True)

In [48]:
df.loc[df.index.get_level_values('sample').str.contains("-11.$")].index

MultiIndex([], names=['subject', 'sample', 'hkle'])

## Drop solitary tumor / normal ( not sure how to do this programmatically )

* FG-5963-01A
* CS-5390-10A

In [49]:
df.loc[df.index.get_level_values('sample') == 'FG-5963-01A'].index

MultiIndex([('FG-5963', 'FG-5963-01A', 'HERVK113'),
            ('FG-5963', 'FG-5963-01A',    'SVA_A'),
            ('FG-5963', 'FG-5963-01A',    'SVA_B'),
            ('FG-5963', 'FG-5963-01A',    'SVA_C'),
            ('FG-5963', 'FG-5963-01A',    'SVA_D'),
            ('FG-5963', 'FG-5963-01A',    'SVA_E'),
            ('FG-5963', 'FG-5963-01A',    'SVA_F')],
           names=['subject', 'sample', 'hkle'])

In [50]:
df.drop(df.loc[df.index.get_level_values('sample') == 'FG-5963-01A'].index,axis='index',inplace=True)

In [51]:
df.loc[df.index.get_level_values('sample') == 'FG-5963-01A'].index

MultiIndex([], names=['subject', 'sample', 'hkle'])

In [52]:
df.loc[df.index.get_level_values('sample') == "CS-5390-10A"].index

MultiIndex([('CS-5390', 'CS-5390-10A', 'HERVK113'),
            ('CS-5390', 'CS-5390-10A',    'SVA_A'),
            ('CS-5390', 'CS-5390-10A',    'SVA_B'),
            ('CS-5390', 'CS-5390-10A',    'SVA_C'),
            ('CS-5390', 'CS-5390-10A',    'SVA_D'),
            ('CS-5390', 'CS-5390-10A',    'SVA_E'),
            ('CS-5390', 'CS-5390-10A',    'SVA_F')],
           names=['subject', 'sample', 'hkle'])

In [53]:
df.drop(df.loc[df.index.get_level_values('sample') == "CS-5390-10A"].index,axis='index',inplace=True)

In [54]:
df.loc[df.index.get_level_values('sample') == "CS-5390-10A"].index

MultiIndex([], names=['subject', 'sample', 'hkle'])

# Match earlier HERVK113 notebook counts
* CS-5390-10A - 553
* FG-5963-01A - 68
* CS-6186-01A - 1100
* CS-6186-10A - 950
* DU-5872-01A - 1108
* DU-5872-10A - 776
* 02-2483-01A - 568
* 02-2483-10A - 557
* 02-2485-01A - 587
* 02-2485-10A - 498
* 06-0125-01A - 734

In [55]:
df[df[df.columns]>0].count(axis='columns')['CS-6186','CS-6186-01A']

hkle
HERVK113    1100
SVA_A       2224
SVA_B       2141
SVA_C       2174
SVA_D       2200
SVA_E       2159
SVA_F       1784
dtype: int64

In [56]:
df[df[df.columns]>0].count(axis='columns')['CS-6186','CS-6186-10A']

hkle
HERVK113     950
SVA_A       1962
SVA_B       1839
SVA_C       1845
SVA_D       1868
SVA_E       1827
SVA_F       1512
dtype: int64

In [57]:
df[df[df.columns]>0].count(axis='columns')['DU-5872','DU-5872-01A']

hkle
HERVK113    1108
SVA_A       2744
SVA_B       2640
SVA_C       2625
SVA_D       2644
SVA_E       2614
SVA_F       2285
dtype: int64

In [58]:
df[df[df.columns]>0].count(axis='columns')['DU-5872','DU-5872-10A']

hkle
HERVK113     776
SVA_A       2008
SVA_B       1904
SVA_C       1908
SVA_D       1926
SVA_E       1884
SVA_F       1547
dtype: int64

## Keep only Broad and drop Harvard Medical School

### Harvard's samples were really low coverage?

In [59]:
meta[meta['center_name'] == 'Broad Institute of MIT and Harvard']['subject'].unique()


array(['02-2483', '02-2485', '06-0125', '06-0152', '06-0157', '06-0171',
       '06-0190', '06-0210', '06-0211', '06-0214', '06-0221', '06-0686',
       '06-0744', '06-0745', '06-2557', '06-2570', '06-5411', '06-5415',
       '14-1034', '14-1402', '14-1823', '14-2554', '15-1444', '19-1389',
       '19-2620', '19-2624', '19-2629', '19-5960', '26-5132', '26-5135',
       '27-1831', '27-2523', '27-2528', '32-1970', '41-5651', 'CS-5390',
       'CS-5395', 'CS-6186', 'CS-6665', 'CS-6668', 'CS-6669', 'DB-5278',
       'DH-A669', 'DU-5870', 'DU-5872', 'DU-5874', 'DU-6397', 'DU-6401',
       'DU-6404', 'DU-6407', 'DU-7009', 'DU-7013', 'DU-7301', 'DU-7304',
       'E1-5318', 'E1-5319', 'EZ-7264', 'FG-5964', 'FG-5965', 'FG-7643',
       'FG-8182', 'FG-A4MT', 'HT-7602', 'HT-7689', 'HT-7695', 'HT-8104',
       'HT-A5R7', 'HT-A61B', 'HW-7486', 'HW-7487', 'IK-7675', 'TM-A7CF',
       'TQ-A7RK', 'TQ-A7RV', 'TQ-A8XE'], dtype=object)

In [60]:
df.index.get_level_values('subject')


Index(['02-2483', '02-2483', '02-2483', '02-2483', '02-2483', '02-2483',
       '02-2483', '02-2483', '02-2483', '02-2483',
       ...
       'TQ-A8XE', 'TQ-A8XE', 'TQ-A8XE', 'TQ-A8XE', 'TQ-A8XE', 'TQ-A8XE',
       'TQ-A8XE', 'TQ-A8XE', 'TQ-A8XE', 'TQ-A8XE'],
      dtype='object', name='subject', length=1708)

In [61]:
df.index.get_level_values('subject').isin(
    meta[meta['center_name'] == 'Broad Institute of MIT and Harvard']['subject'].unique())


array([ True,  True,  True, ...,  True,  True,  True])

In [62]:
df=df[df.index.get_level_values('subject').isin(
    meta[meta['center_name'] == 'Broad Institute of MIT and Harvard']['subject'].unique())]
df.head()


chromosome                      chr1                                          \
position                     1048000 1190000 1308000 1409000 1410000 1580000   
subject sample      hkle                                                       
02-2483 02-2483-01A HERVK113       0       0       0       4       1       4   
                    SVA_A          0       0       0       0       0       0   
                    SVA_B          0       0       0       0       0       0   
                    SVA_C          0       0       0       0       0       0   
                    SVA_D          0       0       0       0       0       0   

chromosome                                                    ...     chrY  \
position                     1581000 2211000 2513000 2581000  ... 19535000   
subject sample      hkle                                      ...            
02-2483 02-2483-01A HERVK113       0       0       0       0  ...        0   
                    SVA_A          0       0       0       0  ...        0   
                    SVA_B          0       0       0       0  ...        0   
                    SVA_C          0       0       0       0  ...        0   
                    SVA_D          0       0       0       0  ...        0   

chromosome                                                                 \
position                     19536000 19699000 19700000 20552000 20553000   
subject sample      hkle                                                    
02-2483 02-2483-01A HERVK113        0        2        0        0        0   
                    SVA_A           0        0        0        0        0   
                    SVA_B           0        0        0        0        0   
                    SVA_C           0        0        0        0        0   
                    SVA_D           0        0        0        0        0   

chromosome                                                        
position                     20976000 21047000 21217000 21639000  
subject sample      hkle                                          
02-2483 02-2483-01A HERVK113        0        0        0        0  
                    SVA_A           0        0        0        0  
                    SVA_B           0        0        0        0  
                    SVA_C           0        0        0        0  
                    SVA_D           0        0        0        0  

[5 rows x 17409 columns]

In [63]:
df.shape

(1022, 17409)

## New technique. Separate the tumor and normal counts.
### Remove the sample column
### DO NOT RESHAPE

In [64]:
n=df.copy()    # NORMAL
t=df.copy()    # TUMOR

In [65]:
n.drop(n.loc[n.index.get_level_values('sample').str.contains("-01.$")].index,axis='index',inplace=True)
t.drop(t.loc[t.index.get_level_values('sample').str.contains("-10.$")].index,axis='index',inplace=True)

In [66]:
n=n.droplevel('sample',axis='index')
t=t.droplevel('sample',axis='index')

In [67]:
n.head()

chromosome          chr1                                                  \
position         1048000 1190000 1308000 1409000 1410000 1580000 1581000   
subject hkle                                                               
02-2483 HERVK113       0       0       0       2       3       3       0   
        SVA_A          0       0       0       0       0       0       0   
        SVA_B          0       0       0       0       0       0       0   
        SVA_C          0       0       0       0       0       0       0   
        SVA_D          0       0       0       0       0       0       0   

chromosome                                ...     chrY                    \
position         2211000 2513000 2581000  ... 19535000 19536000 19699000   
subject hkle                              ...                              
02-2483 HERVK113       0       0       0  ...        0        0        1   
        SVA_A          0       0       0  ...        0        0        0   
        SVA_B          0       0       0  ...        0        0        0   
        SVA_C          0       0       0  ...        0        0        0   
        SVA_D          0       0       0  ...        0        0        0   

chromosome                                                              \
position         19700000 20552000 20553000 20976000 21047000 21217000   
subject hkle                                                             
02-2483 HERVK113        1        0        0        0        0        0   
        SVA_A           0        0        0        0        0        0   
        SVA_B           0        0        0        0        0        0   
        SVA_C           0        0        0        0        0        0   
        SVA_D           0        0        0        0        0        0   

chromosome                 
position         21639000  
subject hkle               
02-2483 HERVK113        0  
        SVA_A           0  
        SVA_B           0  
        SVA_C           0  
        SVA_D           0  

[5 rows x 17409 columns]

In [68]:
t.head()

chromosome          chr1                                                  \
position         1048000 1190000 1308000 1409000 1410000 1580000 1581000   
subject hkle                                                               
02-2483 HERVK113       0       0       0       4       1       4       0   
        SVA_A          0       0       0       0       0       0       0   
        SVA_B          0       0       0       0       0       0       0   
        SVA_C          0       0       0       0       0       0       0   
        SVA_D          0       0       0       0       0       0       0   

chromosome                                ...     chrY                    \
position         2211000 2513000 2581000  ... 19535000 19536000 19699000   
subject hkle                              ...                              
02-2483 HERVK113       0       0       0  ...        0        0        2   
        SVA_A          0       0       0  ...        0        0        0   
        SVA_B          0       0       0  ...        0        0        0   
        SVA_C          0       0       0  ...        0        0        0   
        SVA_D          0       0       0  ...        0        0        0   

chromosome                                                              \
position         19700000 20552000 20553000 20976000 21047000 21217000   
subject hkle                                                             
02-2483 HERVK113        0        0        0        0        0        0   
        SVA_A           0        0        0        0        0        0   
        SVA_B           0        0        0        0        0        0   
        SVA_C           0        0        0        0        0        0   
        SVA_D           0        0        0        0        0        0   

chromosome                 
position         21639000  
subject hkle               
02-2483 HERVK113        0  
        SVA_A           0  
        SVA_B           0  
        SVA_C           0  
        SVA_D           0  

[5 rows x 17409 columns]

In [69]:
n_index=n.index
t_index=t.index

### Compare the indexes. An empty array shows that the indexes are the same. Yay.

In [70]:
n_index[n_index != t_index]

MultiIndex([], names=['subject', 'hkle'])

In [71]:
compdf=(((n>0)^(t>0))&(t>0)).replace([False,True],[0,1]) - (((n>0)^(t>0))&(n>0)).replace([False,True],[0,1])

In [72]:
compdf

chromosome          chr1                                                  \
position         1048000 1190000 1308000 1409000 1410000 1580000 1581000   
subject hkle                                                               
02-2483 HERVK113       0       0       0       0       0       0       0   
        SVA_A          0       0       0       0       0       0       0   
        SVA_B          0       0       0       0       0       0       0   
        SVA_C          0       0       0       0       0       0       0   
        SVA_D          0       0       0       0       0       0       0   
...                  ...     ...     ...     ...     ...     ...     ...   
TQ-A8XE SVA_B          0       0       0       0       0       0       1   
        SVA_C          0       0       0       0       0       0       0   
        SVA_D          0       0       0       0       0       0       0   
        SVA_E          0       0       0       0       0       0       0   
        SVA_F          0       0       0       0       0       0       0   

chromosome                                ...     chrY                    \
position         2211000 2513000 2581000  ... 19535000 19536000 19699000   
subject hkle                              ...                              
02-2483 HERVK113       0       0       0  ...        0        0        0   
        SVA_A          0       0       0  ...        0        0        0   
        SVA_B          0       0       0  ...        0        0        0   
        SVA_C          0       0       0  ...        0        0        0   
        SVA_D          0       0       0  ...        0        0        0   
...                  ...     ...     ...  ...      ...      ...      ...   
TQ-A8XE SVA_B          0       0       0  ...        0        0        0   
        SVA_C          0       0       0  ...        0        0        0   
        SVA_D          0       0       0  ...        0        0        0   
        SVA_E          0       0       0  ...        0        0        0   
        SVA_F          0       0       0  ...        0        0        0   

chromosome                                                              \
position         19700000 20552000 20553000 20976000 21047000 21217000   
subject hkle                                                             
02-2483 HERVK113       -1        0        0        0        0        0   
        SVA_A           0        0        0        0        0        0   
        SVA_B           0        0        0        0        0        0   
        SVA_C           0        0        0        0        0        0   
        SVA_D           0        0        0        0        0        0   
...                   ...      ...      ...      ...      ...      ...   
TQ-A8XE SVA_B           0        0        0        0        0        0   
        SVA_C           0        0        0        0        0        0   
        SVA_D           0        0        0        0        0        0   
        SVA_E           0        0        0        0        0        0   
        SVA_F           0        0        0        0        0        0   

chromosome                 
position         21639000  
subject hkle               
02-2483 HERVK113        0  
        SVA_A           0  
        SVA_B           0  
        SVA_C           0  
        SVA_D           0  
...                   ...  
TQ-A8XE SVA_B           0  
        SVA_C           0  
        SVA_D           0  
        SVA_E           0  
        SVA_F           0  

[511 rows x 17409 columns]

This is cleaner, faster and more pythonic than what follows for compdf.


## Search for the presence of HKLE insertions with positions unique to tumor or normal

### Copy raw dataframe, remove sample level and clear out the values

In [73]:
#compdf=df.copy()
#compdf.drop(compdf.loc[compdf.index.get_level_values('sample').str.contains("-10.$")].index,axis='index',inplace=True)
#compdf=compdf.droplevel('sample',axis='index')
#compdf.loc[:,:] = 0
#compdf.head(10)

## Ok. This is some dark magic I think. Gotta be a better, cleaner way to do this.

* Loop over the subjects and hkles. (Loops aren't really pythony.)
* Do an "XOR-AND" for the first (tumor 01) and second (normal 10) samples.
* This returns a Boolean mask
* Change it to 0/1
* Subtract the tumor array from the normal array and assign it to the subject/hkle

### This will create a dataframe of positions representing "tumor - normal" so with values of only

* -1 : normal only
*  0 : both or neither
* +1 : tumor only


In [74]:
#for subject in df.index.get_level_values('subject').unique():
#    for hkle in df.index.get_level_values('hkle').unique():
#        tm=(((  df.loc[subject,:,hkle].iloc[0]>0)^(
#                df.loc[subject,:,hkle].iloc[1]>0))&(
#                df.loc[subject,:,hkle].iloc[0]>0))
#        tm.replace([False,True],[0,1],inplace=True)
#        nm=(((  df.loc[subject,:,hkle].iloc[0]>0)^(
#                df.loc[subject,:,hkle].iloc[1]>0))&(
#                df.loc[subject,:,hkle].iloc[1]>0))
#        nm.replace([False,True],[0,1],inplace=True)
#        compdf.loc[subject,hkle]=tm.values-nm.values

In [75]:
compdf.head()

chromosome          chr1                                                  \
position         1048000 1190000 1308000 1409000 1410000 1580000 1581000   
subject hkle                                                               
02-2483 HERVK113       0       0       0       0       0       0       0   
        SVA_A          0       0       0       0       0       0       0   
        SVA_B          0       0       0       0       0       0       0   
        SVA_C          0       0       0       0       0       0       0   
        SVA_D          0       0       0       0       0       0       0   

chromosome                                ...     chrY                    \
position         2211000 2513000 2581000  ... 19535000 19536000 19699000   
subject hkle                              ...                              
02-2483 HERVK113       0       0       0  ...        0        0        0   
        SVA_A          0       0       0  ...        0        0        0   
        SVA_B          0       0       0  ...        0        0        0   
        SVA_C          0       0       0  ...        0        0        0   
        SVA_D          0       0       0  ...        0        0        0   

chromosome                                                              \
position         19700000 20552000 20553000 20976000 21047000 21217000   
subject hkle                                                             
02-2483 HERVK113       -1        0        0        0        0        0   
        SVA_A           0        0        0        0        0        0   
        SVA_B           0        0        0        0        0        0   
        SVA_C           0        0        0        0        0        0   
        SVA_D           0        0        0        0        0        0   

chromosome                 
position         21639000  
subject hkle               
02-2483 HERVK113        0  
        SVA_A           0  
        SVA_B           0  
        SVA_C           0  
        SVA_D           0  

[5 rows x 17409 columns]

In [76]:
compdf.shape

(511, 17409)

### Keep only positions with any sample with something other than 0

### We lose over 3000 positions

In [77]:
compdf=compdf.loc[:, (compdf != 0).any(axis=0)]

In [78]:
compdf.shape

(511, 14280)

In [79]:
compdf

chromosome          chr1                                                  \
position         1308000 1409000 1410000 1580000 1581000 2211000 2513000   
subject hkle                                                               
02-2483 HERVK113       0       0       0       0       0       0       0   
        SVA_A          0       0       0       0       0       0       0   
        SVA_B          0       0       0       0       0       0       0   
        SVA_C          0       0       0       0       0       0       0   
        SVA_D          0       0       0       0       0       0       0   
...                  ...     ...     ...     ...     ...     ...     ...   
TQ-A8XE SVA_B          0       0       0       0       1       0       0   
        SVA_C          0       0       0       0       0       0       0   
        SVA_D          0       0       0       0       0       0       0   
        SVA_E          0       0       0       0       0       0       0   
        SVA_F          0       0       0       0       0       0       0   

chromosome                                ...     chrY                    \
position         2581000 3129000 3315000  ... 17181000 19536000 19699000   
subject hkle                              ...                              
02-2483 HERVK113       0       0       0  ...        0        0        0   
        SVA_A          0       0       0  ...        0        0        0   
        SVA_B          0       0       0  ...        0        0        0   
        SVA_C          0       0       0  ...        0        0        0   
        SVA_D          0       0       0  ...        0        0        0   
...                  ...     ...     ...  ...      ...      ...      ...   
TQ-A8XE SVA_B          0       0       0  ...        0        0        0   
        SVA_C          0       0       0  ...        0        0        0   
        SVA_D          0       0       0  ...        0        0        0   
        SVA_E          0       0       0  ...        0        0        0   
        SVA_F          0       0       0  ...        0        0        0   

chromosome                                                              \
position         19700000 20552000 20553000 20976000 21047000 21217000   
subject hkle                                                             
02-2483 HERVK113       -1        0        0        0        0        0   
        SVA_A           0        0        0        0        0        0   
        SVA_B           0        0        0        0        0        0   
        SVA_C           0        0        0        0        0        0   
        SVA_D           0        0        0        0        0        0   
...                   ...      ...      ...      ...      ...      ...   
TQ-A8XE SVA_B           0        0        0        0        0        0   
        SVA_C           0        0        0        0        0        0   
        SVA_D           0        0        0        0        0        0   
        SVA_E           0        0        0        0        0        0   
        SVA_F           0        0        0        0        0        0   

chromosome                 
position         21639000  
subject hkle               
02-2483 HERVK113        0  
        SVA_A           0  
        SVA_B           0  
        SVA_C           0  
        SVA_D           0  
...                   ...  
TQ-A8XE SVA_B           0  
        SVA_C           0  
        SVA_D           0  
        SVA_E           0  
        SVA_F           0  

[511 rows x 14280 columns]

### Get the chromosome columns for analysis after merging with the metadata

In [80]:
chromosome_columns=compdf.columns[compdf.columns.get_level_values('chromosome').str.contains("^chr")]
chromosome_columns

MultiIndex([('chr1',  '1308000'),
            ('chr1',  '1409000'),
            ('chr1',  '1410000'),
            ('chr1',  '1580000'),
            ('chr1',  '1581000'),
            ('chr1',  '2211000'),
            ('chr1',  '2513000'),
            ('chr1',  '2581000'),
            ('chr1',  '3129000'),
            ('chr1',  '3315000'),
            ...
            ('chrY', '17181000'),
            ('chrY', '19536000'),
            ('chrY', '19699000'),
            ('chrY', '19700000'),
            ('chrY', '20552000'),
            ('chrY', '20553000'),
            ('chrY', '20976000'),
            ('chrY', '21047000'),
            ('chrY', '21217000'),
            ('chrY', '21639000')],
           names=['chromosome', 'position'], length=14280)

### Merge with the metadata for later grouping

In [81]:
compdf.reset_index(inplace=True)
compdf=compdf.merge(meta,on='subject')
compdf.set_index(['subject','hkle'],inplace=True)
compdf.head()

/Users/jakewendt/Library/Python/3.6/lib/python/site-packages/pandas/core/generic.py:3889: PerformanceWarning: dropping on a non-lexsorted multi-index without a level parameter may impact performance.
  obj = obj._drop_axis(labels, axis, level=level, errors=errors)


chromosome          chr1                                                  \
position         1308000 1409000 1410000 1580000 1581000 2211000 2513000   
subject hkle                                                               
02-2483 HERVK113       0       0       0       0       0       0       0   
        SVA_A          0       0       0       0       0       0       0   
        SVA_B          0       0       0       0       0       0       0   
        SVA_C          0       0       0       0       0       0       0   
        SVA_D          0       0       0       0       0       0       0   

chromosome                                ...  \
position         2581000 3129000 3315000  ...   
subject hkle                              ...   
02-2483 HERVK113       0       0       0  ...   
        SVA_A          0       0       0  ...   
        SVA_B          0       0       0  ...   
        SVA_C          0       0       0  ...   
        SVA_D          0       0       0  ...   

chromosome                              Triple_group  \
position                                               
subject hkle                                           
02-2483 HERVK113  IDH-Mutant:1p19q-non-codel:TERT-WT   
        SVA_A     IDH-Mutant:1p19q-non-codel:TERT-WT   
        SVA_B     IDH-Mutant:1p19q-non-codel:TERT-WT   
        SVA_C     IDH-Mutant:1p19q-non-codel:TERT-WT   
        SVA_D     IDH-Mutant:1p19q-non-codel:TERT-WT   

chromosome           Tissue_sample_location        MGMT   Age Survival_months  \
position                                                                        
subject hkle                                                                    
02-2483 HERVK113  MD Anderson Cancer Center  Methylated  43.0       15.310383   
        SVA_A     MD Anderson Cancer Center  Methylated  43.0       15.310383   
        SVA_B     MD Anderson Cancer Center  Methylated  43.0       15.310383   
        SVA_C     MD Anderson Cancer Center  Methylated  43.0       15.310383   
        SVA_D     MD Anderson Cancer Center  Methylated  43.0       15.310383   

chromosome       Vital_status days_to_death gender1             disease_type  \
position                                                                       
subject hkle                                                                   
02-2483 HERVK113          0.0           NaN    male  Glioblastoma Multiforme   
        SVA_A             0.0           NaN    male  Glioblastoma Multiforme   
        SVA_B             0.0           NaN    male  Glioblastoma Multiforme   
        SVA_C             0.0           NaN    male  Glioblastoma Multiforme   
        SVA_D             0.0           NaN    male  Glioblastoma Multiforme   

chromosome                               center_name  
position                                              
subject hkle                                          
02-2483 HERVK113  Broad Institute of MIT and Harvard  
        SVA_A     Broad Institute of MIT and Harvard  
        SVA_B     Broad Institute of MIT and Harvard  
        SVA_C     Broad Institute of MIT and Harvard  
        SVA_D     Broad Institute of MIT and Harvard  

[5 rows x 14301 columns]

### More dark magic here. 

* Select all subjects and HERVK113 on the y-axis and all the chromosome columns on the x-axis.
* We will count all the incidence of each of the previously computed "tumor-normal" values
* Then we will transpose and sort descending by "-1". This will put the highest occurence of normal at the top.


In [82]:
compdf.loc[pd.IndexSlice[:,'HERVK113'],chromosome_columns].apply(
    pd.Series.value_counts, axis=0).T.sort_values(by=[-1],ascending=False).head(10)


-1     0     1
chromosome position                   
chr10      121929000  21.0  34.0  18.0
chr20      29440000   21.0  36.0  16.0
chr21      17932000   20.0  30.0  23.0
chr18      50404000   20.0  27.0  26.0
chr19      51924000   20.0  37.0  16.0
chr5       120194000  20.0  31.0  22.0
chr9       17446000   19.0  35.0  19.0
chr2       208257000  19.0  35.0  19.0
chr3       118539000  19.0  34.0  20.0
chr19      53497000   19.0  41.0  13.0

In [83]:
compdf.loc[pd.IndexSlice[:,'SVA_A'],chromosome_columns].apply(
    pd.Series.value_counts, axis=0).T.sort_values(by=[-1],ascending=False).head(10)


-1     0     1
chromosome position                   
chr13      45484000   23.0  36.0  14.0
chr15      80648000   22.0  35.0  16.0
chr10      34977000   22.0  29.0  22.0
chr17      66269000   21.0  37.0  15.0
chr14      31559000   21.0  36.0  16.0
chr3       196944000  21.0  39.0  13.0
chr2       64041000   21.0  35.0  17.0
chr1       181955000  21.0  34.0  18.0
chr8       117428000  20.0  33.0  20.0
chr10      24658000   20.0  41.0  12.0

In [84]:
compdf.loc[pd.IndexSlice[:,'SVA_B'],chromosome_columns].apply(
    pd.Series.value_counts, axis=0).T.sort_values(by=[-1],ascending=False).head(10)


-1     0     1
chromosome position                   
chr18      70570000   25.0  31.0  17.0
chr22      11550000   23.0  33.0  17.0
chr10      75751000   23.0  39.0  11.0
chr5       171148000  23.0  31.0  19.0
chr10      32232000   23.0  40.0  10.0
chr18      54342000   22.0  36.0  15.0
chr21      28563000   22.0  40.0  11.0
chr10      92673000   22.0  39.0  12.0
chr18      39511000   21.0  35.0  17.0
chr2       155658000  21.0  33.0  19.0

In [85]:
compdf.loc[pd.IndexSlice[:,'SVA_C'],chromosome_columns].apply(
    pd.Series.value_counts, axis=0).T.sort_values(by=[-1],ascending=False).head(10)


-1     0     1
chromosome position                   
chr18      70570000   25.0  30.0  18.0
chr10      75751000   23.0  37.0  13.0
chr13      21449000   22.0  31.0  20.0
chr21      28563000   22.0  40.0  11.0
chr4       6978000    22.0  36.0  15.0
chr6       18227000   22.0  31.0  20.0
chr8       128824000  21.0  29.0  23.0
chr18      39511000   21.0  35.0  17.0
chr9       38325000   21.0  36.0  16.0
chr18      54342000   21.0  37.0  15.0

In [86]:
compdf.loc[pd.IndexSlice[:,'SVA_D'],chromosome_columns].apply(
    pd.Series.value_counts, axis=0).T.sort_values(by=[-1],ascending=False).head(10)


-1     0     1
chromosome position                   
chr18      70570000   25.0  31.0  17.0
chr10      75751000   23.0  37.0  13.0
chr4       6978000    22.0  36.0  15.0
chr21      28563000   22.0  40.0  11.0
chr13      21449000   22.0  31.0  20.0
chr6       18227000   22.0  31.0  20.0
chr8       128824000  21.0  29.0  23.0
chr18      39511000   21.0  35.0  17.0
chr9       38325000   21.0  36.0  16.0
chr18      54342000   21.0  37.0  15.0

In [87]:
compdf.loc[pd.IndexSlice[:,'SVA_E'],chromosome_columns].apply(
    pd.Series.value_counts, axis=0).T.sort_values(by=[-1],ascending=False).head(10)


-1     0     1
chromosome position                   
chr18      70570000   25.0  30.0  18.0
chr10      118894000  24.0  35.0  14.0
chr6       18227000   22.0  31.0  20.0
chr4       6978000    22.0  36.0  15.0
chr13      21449000   22.0  31.0  20.0
chr10      75751000   22.0  39.0  12.0
chr21      28563000   22.0  40.0  11.0
chr2       155658000  21.0  33.0  19.0
chr13      32715000   21.0  35.0  17.0
chr18      39511000   21.0  35.0  17.0

In [88]:
compdf.loc[pd.IndexSlice[:,'SVA_F'],chromosome_columns].apply(
    pd.Series.value_counts, axis=0).T.sort_values(by=[-1],ascending=False).head(10)


-1     0     1
chromosome              position                   
chr1                    32803000   23.0  34.0  16.0
chr18                   70570000   23.0  33.0  17.0
chr19                   38856000   22.0  41.0  10.0
chr14_GL000194v1_random 189000     22.0  42.0   9.0
chr10                   75751000   22.0  45.0   6.0
chr3                    118717000  21.0  43.0   9.0
chr6                    69434000   21.0  36.0  16.0
chr19                   45649000   21.0  38.0  14.0
chr2                    36181000   21.0  28.0  24.0
chr22                   49029000   21.0  38.0  14.0

We see that the unique normal positions aren't really that unique.

If we do the same, but sort by tumor ...

In [89]:
compdf.loc[pd.IndexSlice[:,'HERVK113'],chromosome_columns].apply(
    pd.Series.value_counts, axis=0).T.sort_values(by=[1],ascending=False).head(10)


-1     0     1
chromosome position                   
chr19      37629000   10.0  34.0  29.0
           27814000   10.0  34.0  29.0
chrX       113990000  13.0  31.0  29.0
chr19      23078000   16.0  29.0  28.0
chr1       160914000  11.0  34.0  28.0
chr9       111876000  15.0  31.0  27.0
chr20      7957000     8.0  38.0  27.0
chr7       99432000    6.0  40.0  27.0
chr19      52420000   10.0  36.0  27.0
           57557000   14.0  32.0  27.0

In [90]:
compdf.loc[pd.IndexSlice[:,'SVA_A'],chromosome_columns].apply(
    pd.Series.value_counts, axis=0).T.sort_values(by=[1],ascending=False).head(10)


-1     0     1
chromosome position                   
chr7       6976000     8.0  34.0  31.0
           92544000    5.0  37.0  31.0
chr2       34132000    7.0  36.0  30.0
chr19      6513000     7.0  37.0  29.0
chrX       40863000   14.0  30.0  29.0
chr19      20953000    9.0  35.0  29.0
chr22      42972000   14.0  31.0  28.0
chr1       180710000  12.0  33.0  28.0
chr6       30221000   12.0  33.0  28.0
chr19      23160000   11.0  34.0  28.0

In [91]:
compdf.loc[pd.IndexSlice[:,'SVA_B'],chromosome_columns].apply(
    pd.Series.value_counts, axis=0).T.sort_values(by=[1],ascending=False).head(10)


-1     0     1
chromosome position                   
chr7       91886000   10.0  30.0  33.0
chr6       30221000    8.0  34.0  31.0
chr7       77510000   11.0  31.0  31.0
chr2       27558000    7.0  36.0  30.0
chr11      95481000    9.0  34.0  30.0
chr5       59464000   12.0  31.0  30.0
chr7       54675000    6.0  37.0  30.0
chr8       128824000  18.0  25.0  30.0
chr3       113047000   6.0  38.0  29.0
chr12      110173000  10.0  34.0  29.0

In [92]:
compdf.loc[pd.IndexSlice[:,'SVA_C'],chromosome_columns].apply(
    pd.Series.value_counts, axis=0).T.sort_values(by=[1],ascending=False).head(10)


-1     0     1
chromosome position                   
chr19      6513000     4.0  36.0  33.0
chr7       91886000   10.0  30.0  33.0
chr8       127317000  11.0  31.0  31.0
chr2       27558000    7.0  36.0  30.0
chr11      95481000    8.0  36.0  29.0
chr14      23107000   10.0  34.0  29.0
chr6       30221000   11.0  33.0  29.0
chr12      110173000   9.0  35.0  29.0
chr2       98989000    8.0  36.0  29.0
chr7       77510000   11.0  34.0  28.0

In [93]:
compdf.loc[pd.IndexSlice[:,'SVA_D'],chromosome_columns].apply(
    pd.Series.value_counts, axis=0).T.sort_values(by=[1],ascending=False).head(10)


-1     0     1
chromosome position                   
chr19      6513000     4.0  36.0  33.0
chr7       91886000   11.0  29.0  33.0
chr8       127317000  11.0  31.0  31.0
chr2       27558000    7.0  36.0  30.0
chr11      95481000    9.0  35.0  29.0
chr6       30221000   11.0  33.0  29.0
chr18      7203000    11.0  33.0  29.0
chr14      23107000   10.0  34.0  29.0
chr12      110173000   9.0  35.0  29.0
chr7       27986000   10.0  35.0  28.0

In [94]:
compdf.loc[pd.IndexSlice[:,'SVA_E'],chromosome_columns].apply(
    pd.Series.value_counts, axis=0).T.sort_values(by=[1],ascending=False).head(10)


-1     0     1
chromosome position                   
chr19      6513000     4.0  36.0  33.0
chr7       91886000   11.0  30.0  32.0
chr8       127317000  11.0  31.0  31.0
chr11      95481000   10.0  32.0  31.0
chr12      110173000  11.0  32.0  30.0
chr2       27558000    7.0  37.0  29.0
chr6       30221000   11.0  33.0  29.0
chr18      7203000    11.0  33.0  29.0
chr14      23107000   10.0  34.0  29.0
chr7       27986000    9.0  36.0  28.0

In [95]:
compdf.loc[pd.IndexSlice[:,'SVA_F'],chromosome_columns].apply(
    pd.Series.value_counts, axis=0).T.sort_values(by=[1],ascending=False).head(10)


-1     0     1
chromosome position                   
chr19      6513000     7.0  32.0  34.0
chr7       91886000   11.0  29.0  33.0
chr17      35194000    8.0  35.0  30.0
chr7       16789000   10.0  33.0  30.0
chr22      42972000   16.0  27.0  30.0
chr2       33760000    8.0  35.0  30.0
chr12      110173000  10.0  34.0  29.0
           6474000    10.0  34.0  29.0
chr11      95481000    9.0  35.0  29.0
chr20      31428000    6.0  38.0  29.0

### These values seem to be a bit more lopsided.

Further analysis forthcoming.

In [96]:
stopohere

NameError: name 'stopohere' is not defined

# Get raw read counts to normalize

In [ ]:
c=pd.read_csv("TCGA/TCGA_raw_counts.csv")
c.rename(columns={"subject": "sample"},inplace=True)
#c.set_index('sample',inplace=True)
c.head()

In [ ]:
#c.loc['CS-6665-01A','raw pair count']

In [ ]:
#c.loc['CS-6665-10A','raw pair count']

In [ ]:
#df.head()

In [ ]:
#c.columns=[c.columns,['''',]]
c.columns=[c.columns,[''] * len(c.columns)]

In [ ]:
c.head()

In [ ]:
df.reset_index(inplace=True)

In [ ]:
df=df.merge(c,on='sample')

Not sure if this has any impact.

/Users/jakewendt/Library/Python/3.6/lib/python/site-packages/pandas/core/generic.py:3889: PerformanceWarning: dropping on a non-lexsorted multi-index without a level parameter may impact performance.
  obj = obj._drop_axis(labels, axis, level=level, errors=errors)


In [ ]:
df.head()

In [ ]:
df.set_index(['subject','sample','hkle'],inplace=True)
#df.set_index(['subject','sample','hkle','raw pair count'],inplace=True)

In [ ]:
df.head()

# Normalize (not sure if necessary here)

In [ ]:
#dfn=df.copy()

In [ ]:
#dfn.shape

In [ ]:
df.columns

In [ ]:
chromosome_columns=df.columns[df.columns.get_level_values('chromosome').str.contains("^chr")]
chromosome_columns

In [ ]:
df.loc['CS-6665','CS-6665-01A']

In [ ]:
df.loc['CS-6665','CS-6665-10A']

In [ ]:
#df[chromosome_columns]=(1e9*df[chromosome_columns].values)/df[['raw pair count']].values

In [ ]:
df.head()

In [ ]:
df.drop("raw pair count",axis='columns',inplace=True)

In [ ]:
df.head()

In [ ]:
df.shape

In [ ]:
df.loc['CS-6665','CS-6665-01A']

In [ ]:
df.loc['CS-6665','CS-6665-10A']

#  Separate individual HERVs and SVAs

In [ ]:
#hervk113=df.loc[pd.IndexSlice[:,:,'HERVK113'],:]
#sva_a=df.loc[pd.IndexSlice[:,:,'SVA_A'],:]
#sva_b=df.loc[pd.IndexSlice[:,:,'SVA_B'],:]
#sva_c=df.loc[pd.IndexSlice[:,:,'SVA_C'],:]
#sva_d=df.loc[pd.IndexSlice[:,:,'SVA_D'],:]
#sva_e=df.loc[pd.IndexSlice[:,:,'SVA_E'],:]
#sva_f=df.loc[pd.IndexSlice[:,:,'SVA_F'],:]

hervk113=df.loc[df.index.get_level_values('hkle') == "HERVK113",:]
sva_a=df.loc[df.index.get_level_values('hkle') == 'SVA_A',:]
sva_b=df.loc[df.index.get_level_values('hkle') == 'SVA_B',:]
sva_c=df.loc[df.index.get_level_values('hkle') == 'SVA_C',:]
sva_d=df.loc[df.index.get_level_values('hkle') == 'SVA_D',:]
sva_e=df.loc[df.index.get_level_values('hkle') == 'SVA_E',:]
sva_f=df.loc[df.index.get_level_values('hkle') == 'SVA_F',:]

In [ ]:
hervk113.head()

In [ ]:
hervk113.shape

# Now the HKLE index is useless

In [ ]:
#df=df.droplevel('hkle',axis='columns')

In [ ]:
#df.head()

In [ ]:
#df.shape

In [ ]:
hervk113=hervk113.droplevel('hkle',axis='index')
sva_a=sva_a.droplevel('hkle',axis='index')
sva_b=sva_b.droplevel('hkle',axis='index')
sva_c=sva_c.droplevel('hkle',axis='index')
sva_d=sva_d.droplevel('hkle',axis='index')
sva_e=sva_e.droplevel('hkle',axis='index')
sva_f=sva_f.droplevel('hkle',axis='index')

#for ldf in hervk113,sva_a,sva_b,sva_c,sva_d,sva_e,sva_f:
#    ldf=ldf.droplevel('hkle',axis='index')


In [ ]:
hervk113.head()

In [ ]:
hervk113.shape

# Drop all those with 0 insertions

In [ ]:
#for ldf in hervk113,sva_a,sva_b,sva_c,sva_d,sva_e,sva_f:
#    ldf.drop(ldf.loc[:,ldf.sum(axis='index')==0].columns,axis='columns',inplace=True)
    
hervk113.drop(hervk113.loc[:,hervk113.sum(axis='index')==0].columns,axis='columns',inplace=True)
sva_a.drop(sva_a.loc[:,sva_a.sum(axis='index')==0].columns,axis='columns',inplace=True)
sva_b.drop(sva_b.loc[:,sva_b.sum(axis='index')==0].columns,axis='columns',inplace=True)
sva_c.drop(sva_c.loc[:,sva_c.sum(axis='index')==0].columns,axis='columns',inplace=True)
sva_d.drop(sva_d.loc[:,sva_d.sum(axis='index')==0].columns,axis='columns',inplace=True)
sva_e.drop(sva_d.loc[:,sva_e.sum(axis='index')==0].columns,axis='columns',inplace=True)
sva_f.drop(sva_f.loc[:,sva_f.sum(axis='index')==0].columns,axis='columns',inplace=True)


In [ ]:
hervk113.head()

In [ ]:
hervk113.shape

In [ ]:
sva_a.shape

In [ ]:
sva_b.shape

In [ ]:
sva_c.shape

In [ ]:
sva_d.shape

In [ ]:
sva_d.shape

In [ ]:
sva_f.shape

# How to compare the precise site presence in tumor / normal pairs?

In [ ]:
h=hervk113.copy()
h.head()

In [ ]:
for subject in hervk113.index.get_level_values('subject'):
    h.loc[subject]=[
        ((( hervk113.loc[subject].iloc[0]>0)^(
            hervk113.loc[subject].iloc[1]>0))&(
            hervk113.loc[subject].iloc[0]>0)),
        ((( hervk113.loc[subject].iloc[0]>0)^(
            hervk113.loc[subject].iloc[1]>0))&(
            hervk113.loc[subject].iloc[1]>0))]

In [ ]:
h.head()

In [ ]:
hervk113[h].head()

NaN means that neither tumor nor normal were unique at that position.

sum==0 mean that position is never unique.

In [ ]:
#hervk113.loc[:,hervk113[h].fillna(0).sum(axis='index')==0].head()

In [ ]:
#hervk113.loc[:,hervk113[h].fillna(0).sum(axis='index')>0].head()

In [ ]:
#hervk113.loc[:,hervk113[h].fillna(0).sum(axis='index')>0]

In [ ]:
#(hervk113[h]/hervk113[h]).fillna(0).sum(axis='index')==0

In [ ]:
#hervk113.index.get_level_values('subject')

In [ ]:
#hervk113.columns

In [ ]:
htest=pd.DataFrame(columns=hervk113.columns,index=hervk113.index.get_level_values('subject').unique())

In [ ]:
for subject in htest.index.get_level_values('subject'):
    tm=(((  hervk113.loc[subject].iloc[0]>0)^(
            hervk113.loc[subject].iloc[1]>0))&(
            hervk113.loc[subject].iloc[0]>0))
    tm.replace([False,True],[0,1],inplace=True)
    nm=(((  hervk113.loc[subject].iloc[0]>0)^(
            hervk113.loc[subject].iloc[1]>0))&(
            hervk113.loc[subject].iloc[1]>0))
    nm.replace([False,True],[0,1],inplace=True)
    htest.loc[subject]=tm.values-nm.values

In [ ]:
htest

In [ ]:
#hervk113.drop(hervk113.loc[:,hervk113.sum(axis='index')==0].columns,axis='columns',inplace=True)
htest.loc[:,htest.count(axis='index')==0].columns

In [ ]:
#(htest == 0).all(axis=0)

In [ ]:
(htest != 0).any(axis=0)

In [ ]:
# Remove any 0-only positions
#tmp=htest.loc[:, (htest != 0).any(axis=0)]

In [ ]:
htest

In [ ]:
htest.apply(pd.Series.value_counts, axis=0)

tumor - normal so ...

* -1 : normal only
*  0 : both
* +1 : tumor only



In [ ]:
pd.set_option('display.max_rows', None)

In [ ]:
htest.apply(pd.Series.value_counts, axis=0).T.sort_values(by=[-1],ascending=False).head(10)

In [ ]:
htest.apply(pd.Series.value_counts, axis=0).T.sort_values(by=[-1],ascending=True).head(10)

In [ ]:
htest.apply(pd.Series.value_counts, axis=0).T.sort_values(by=[1],ascending=False).head(10)

In [ ]:
htest.apply(pd.Series.value_counts, axis=0).T.sort_values(by=[1,-1,0],ascending=True).head(10)

In [ ]:
htest.apply(pd.Series.value_counts, axis=0).T.sort_values(by=[0],ascending=True).head(10)

In [ ]:
pd.set_option('display.max_rows', max_rows)

In [ ]:
stop before here

In [ ]:
##dftns.loc['TQ-A8XE',0:,:]>0

##dftns.loc[idx['TQ-A8XE',1,:]]
##dftns.loc['TQ-A8XE'].index.get_level_values(0)[0]

#dftns.loc['TQ-A8XE',:,:]>0
dftns.loc['TQ-A8XE',:]>0

In [ ]:
#dftns.loc['TQ-A8XE','TQ-A8XE-01A',:]>0
dftns.loc['TQ-A8XE','TQ-A8XE-01A']>0

In [ ]:
#dftns.loc['TQ-A8XE','TQ-A8XE-10A',:]>0
dftns.loc['TQ-A8XE','TQ-A8XE-10A']>0

# OR

In [ ]:
#(dftns.loc['TQ-A8XE','TQ-A8XE-01A',724443745]>0)|(dftns.loc['TQ-A8XE','TQ-A8XE-10A',:]>0)

(dftns.loc['TQ-A8XE','TQ-A8XE-01A']>0)|(dftns.loc['TQ-A8XE','TQ-A8XE-10A']>0)


# AND

In [ ]:
(dftns.loc['TQ-A8XE','TQ-A8XE-01A']>0)&(dftns.loc['TQ-A8XE','TQ-A8XE-10A']>0)


# A XOR B ( means only A or B, never both )

In [ ]:
(dftns.loc['TQ-A8XE','TQ-A8XE-01A']>0)^(dftns.loc['TQ-A8XE','TQ-A8XE-10A']>0)


# ( A XOR B ) AND A ( means only A )

# ( A XOR B ) AND B ( means only B )


In [ ]:
((dftns.loc['TQ-A8XE','TQ-A8XE-01A']>0)^(dftns.loc['TQ-A8XE','TQ-A8XE-10A']>0))&(dftns.loc['TQ-A8XE','TQ-A8XE-01A']>0)


In [ ]:
((dftns.loc['TQ-A8XE','TQ-A8XE-01A']>0)^(dftns.loc['TQ-A8XE','TQ-A8XE-10A']>0))&(dftns.loc['TQ-A8XE','TQ-A8XE-10A']>0)


In [ ]:
(((dftns.loc['TQ-A8XE','TQ-A8XE-01A']>0)^(dftns.loc['TQ-A8XE','TQ-A8XE-10A']>0))&(dftns.loc['TQ-A8XE','TQ-A8XE-01A']>0)).sum()


In [ ]:
(((dftns.loc['TQ-A8XE','TQ-A8XE-01A']>0)^(dftns.loc['TQ-A8XE','TQ-A8XE-10A']>0))&(dftns.loc['TQ-A8XE','TQ-A8XE-10A']>0)).sum()


In [ ]:
dftns.loc['TQ-A8XE'].iloc[0]

In [ ]:
dftns.loc['TQ-A8XE'].iloc[1]

In [ ]:
dftns.index.levels[0]

In [ ]:
dftns.index.get_level_values(0)

In [ ]:
(dftns.loc['TQ-A8XE'].iloc[0]>0).sum()

In [ ]:
#np.set_printoptions(threshold=np.inf)

In [ ]:
dftns.loc['TQ-A8XE'].iloc[0].values

In [ ]:
#np.set_printoptions(threshold=1000)

In [ ]:
max_rows=pd.get_option('display.max_rows')

In [ ]:
#pd.set_option('display.max_rows', None)

In [ ]:
dftns.head()

In [ ]:
#pd.DataFrame(df[(df['diff']>10) | (df['diff']<-10)].index)

In [ ]:
pd.set_option('display.max_rows', max_rows)

In [ ]:
stn=pd.DataFrame(columns=['subject','uniq tumor','uniq normal'])
stn.set_index('subject',inplace=True)

sstu=pd.DataFrame(columns=['subject','sample','tn','uniq'])
sstu.set_index(['subject','sample','tn'],inplace=True)


In [ ]:
for i in dftns.index.levels[0]:
    stn.loc[i]=((((dftns.loc[i].iloc[0]>0)^(dftns.loc[i].iloc[1]>0))&(dftns.loc[i].iloc[0]>0)).sum(),
               (((dftns.loc[i].iloc[0]>0)^(dftns.loc[i].iloc[1]>0))&(dftns.loc[i].iloc[1]>0)).sum())
    sstu.loc[(i,dftns.loc[i].iloc[0].name, 'tumor'),:]=(
        ((dftns.loc[i].iloc[0]>0)^(dftns.loc[i].iloc[1]>0))&(dftns.loc[i].iloc[0]>0)).sum()
    sstu.loc[(i,dftns.loc[i].iloc[1].name,'normal'),:]=(
        ((dftns.loc[i].iloc[0]>0)^(dftns.loc[i].iloc[1]>0))&(dftns.loc[i].iloc[1]>0)).sum()

# Plot tries to parse the subject ids into dates because they are ##-####
# I can't figure out how to stop it.
#    abctn.loc[("_"+i,dftns.loc[i].iloc[0].name, 'tumor'),:]=(
#((dftns.loc[i].iloc[0]>0)^(dftns.loc[i].iloc[1]>0))&(dftns.loc[i].iloc[0]>0)).sum()
#    abctn.loc[("_"+i,dftns.loc[i].iloc[1].name,'normal'),:]=(
#((dftns.loc[i].iloc[0]>0)^(dftns.loc[i].iloc[1]>0))&(dftns.loc[i].iloc[1]>0)).sum()
    

In [ ]:
dftns.loc['02-2483'].iloc[1].name

In [ ]:
stn

In [ ]:
sstu

In [ ]:
stn.plot(kind='bar',y=['uniq normal','uniq tumor'])
plt.xticks(rotation=75,ha='right')
plt.show()

In [ ]:
sstu.head()

In [ ]:
sstu.reset_index(inplace=True)

In [ ]:
sstu.head()

In [ ]:
groups = sstu.groupby('tn')
fig, ax = plt.subplots()
plt.xticks(rotation=90)
for name, group in groups:
    ax.bar(group['sample'],group['uniq'], label=name)
#    ax.bar(group['subject'],group['uniq'], label=name)
ax.legend()
plt.show()

In [ ]:
meta.head()

In [ ]:
sstum=sstu.merge(meta,on='subject')

In [ ]:
sstum.head()

In [ ]:
groups = sstum.groupby('disease_type')
fig, ax = plt.subplots()
plt.xticks(rotation=90)
for name, group in groups:
    ax.bar(group['sample'],group['uniq'], label=name)
ax.legend()
plt.show()

In [ ]:
sstum.boxplot(column='uniq',
    by=['disease_type','tn'])

In [ ]:
sstum.boxplot(column='uniq',
    by=['IDH','tn'])

In [ ]:
sstum.boxplot(column='uniq',
    by=['TERT','tn'])

In [ ]:
sstum.boxplot(column='uniq',
    by=['x1p19q','tn'])

In [ ]:
sstum.boxplot(column='uniq',
    by=['disease_type','tn','IDH','TERT','x1p19q'])
plt.xticks(rotation=75,ha='right')
plt.show()

In [ ]:
sstum.set_index(['subject','sample','tn'],inplace=True)

In [ ]:
sstum.head()

# Simple sample / tumor count / normal count / disease type plots

In [ ]:
stnm=stn.merge(meta,left_index=True,right_on='subject')

In [ ]:
stnm.set_index('subject',inplace=True)

In [ ]:
stnm.index

In [ ]:
stnm.head()

In [ ]:
stnm.groupby('disease_type').plot(kind='bar',y=['uniq normal','uniq tumor'])


# END Simple sample / tumor count / normal count / disease type plots

In [ ]:
#sstu.reset_index(inplace=True)
#sstu.head()

In [ ]:
#sstum=sstu.merge(meta,on='subject')

In [ ]:
#sstum.head()

# Transpose again

In [ ]:
dftnt=dftn.T

In [ ]:
dftnt.head()

# Number of sites for each subject

In [ ]:
dftnt['normal_count']=dftnt.loc[:,
    dftnt.columns.str.match('.*-10.$')][ dftnt.loc[:,dftnt.columns.str.match('.*-10.$')]>1 ].count(axis=1)

In [ ]:
dftnt['normal_count'].plot(kind='bar')

In [ ]:
dftnt['tumor_count']=dftnt.loc[:,
    dftnt.columns.str.match('.*-01.$')][ dftnt.loc[:,dftnt.columns.str.match('.*-01.$')]>1 ].count(axis=1)

In [ ]:
dftnt['tumor_count'].plot(kind='bar')

# Diff is normal - tumor. Negative means more in tumor.

In [ ]:
dftnt['diff']=dftnt['normal_count']-dftnt['tumor_count']

In [ ]:
dftnt['diff']

In [ ]:
dftnt['diff'].plot(kind='bar')

In [ ]:
dftnt[(dftnt['diff']>5) | (dftnt['diff']<-5)]['diff'].plot(kind='bar')

In [ ]:
dftnt[(dftnt['diff']>10) | (dftnt['diff']<-10)]['diff'].plot(kind='bar')
plt.xticks(rotation=75,ha='right')
plt.show()

In [ ]:
initial_max_rows=pd.get_option('display.max_rows')

In [ ]:
pd.set_option('display.max_rows', None)

In [ ]:
pd.DataFrame(index=dftnt[(dftnt['diff']>10) | (dftnt['diff']<-10)].index,
             data=dftnt[['diff','normal_count','tumor_count']])


In [ ]:
pd.set_option('display.max_rows', initial_max_rows)

In [ ]:
#df.loc[:,df.columns.str.match('.*-01.$')].shape

In [ ]:
#tumor=df.loc[:,df.columns.str.match('.*-01.$')]

In [ ]:
#tumor.shape

In [ ]:
#df.loc[:,df.columns.str.match('.*-10.$')].shape

In [ ]:
#normal=df.loc[:,df.columns.str.match('.*-10.$')]

In [ ]:
#normal.shape

In [ ]:
#normal[normal>0].count(axis=1)

In [ ]:
#normal.loc[:,normal.loc['chr1',1409000]>0]

In [ ]:
#tumor.loc[:,tumor.loc['chr1',1409000]>0]

In [ ]:
#len(normal.loc[:,normal.loc['chr1',1409000]>0].columns)

In [ ]:
#df['normal_count']=df.loc[:,
#    df.columns.str.match('.*-10.$')][ df.loc[:,df.columns.str.match('.*-10.$')]>0 ].count(axis=1)

In [ ]:
#df['tumor_count']=df.loc[:,
#    df.columns.str.match('.*-01.$')][ df.loc[:,df.columns.str.match('.*-01.$')]>0 ].count(axis=1)

In [ ]:
#df['diff']=df['normal_count']-df['tumor_count']

In [ ]:
#df['diff']

In [ ]:
#df['diff'].plot(kind='bar')

In [ ]:
#df[(df['diff']>10) | (df['diff']<-10)]['diff'].plot(kind='bar')

In [ ]:
pd.set_option('display.max_rows', None)

In [ ]:
#pd.DataFrame(df[(df['diff']>10) | (df['diff']<-10)].index)

In [ ]:
pd.set_option('display.max_rows', max_rows)